server/src/main/java/org/opensearch/search/query_group/QueryGroupServiceSettings.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.query_group;
+
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+
+/**
+ * Main class to declare the QueryGroup feature related settings
+ */
+public class QueryGroupServiceSettings {
```

Comment:
@kaushalmahi12 @ruai0511 - I am not sure if `QueryGroupServiceSettings` is the correct name. Probably, `WorkloadManagementSettings` or `ResourceBasedCancellationSettings`? or some other suggestion?

Reply:
I think `CancellationSettings` should be enough or `WorkloadManagementSettings` either is fine but `ResourceBasedCancellationSettings ` seems long and little misnomer given that SBP is also does same sort of thing